pkg/conf/conf_test.go

```diff
@@ -0,0 +1,41 @@
+package conf
+
+import (
+	"testing"
+)
+
+func TestGetPrometheusExporterPorts(t *testing.T) {
+	path := "testdata/testConfig.yml"
+	err := LoadHarvestConfig(path)
+	// Test without checking
+	ValidatePortInUse = true
+	if err != nil {
+		panic(err)
+	}
+	type args struct {
+		pollerName string
+	}
+
+	type test struct {
+		name    string
+		args    args
+		wantErr bool
+	}
+	tests := []test{
+		test{"success", args{pollerName: "unix"}, false},
+		test{"failure_no_free_port", args{pollerName: "cluster-01"}, true},
+		test{"failure_poller_name_does_not_exist", args{pollerName: "test1"}, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetPrometheusExporterPorts(tt.args.pollerName)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetPrometheusExporterPorts() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
```

Comment:
not clear about this, if wantErr is true, then it would show error if actual error comes ?

Reply:
no if we get error then test cases pass